### PR TITLE
Wip/usb msc dynamic init

### PR DESCRIPTION
**Pull request by nickstoughton**
This PR brings in upstream 2.6.1 changes as well as updates to the USB subsys / drivers to better support High Speed

### CODE REVIEW

File: drivers/usb/device/usb_dc_stm32.c [hunk 1]
```diff
@@ -112,8 +112,22 @@ static const struct soc_gpio_pinctrl usb_pinctrl[] =
...
+/* The RX FIFO gets whatever is left */
+#define FIFO_RX_WORDS	(FIFO_WORDS - FIFO_NUM_LG*FIFO_LG_TX_WORDS - \
+		FIFO_NUM_SM*FIFO_SM_TX_WORDS)
+/*
+ * Device RxFIFO =
+ * (5 * number of control endpoints + 8) + ((largest USB packet used / 4) + 1 for status
+ * information) + (2 * number of OUT endpoints) + 1 for Global NAK
+ */
+BUILD_ASSERT(FIFO_RX_WORDS >= (5 * 1 + 8) + ((EP_MPS / 4) + 1) + (2 * 2) + 1,
+		"Recevive FIFO too small");
```
Comment (djcb) [inline drivers/usb/device/usb_dc_stm32.c:131]: Receive

File: drivers/usb/device/usb_dc_stm32.c [hunk 2]
```diff
@@ -112,8 +112,22 @@ static const struct soc_gpio_pinctrl usb_pinctrl[] =
 /* 4-byte words FIFO */
 #define FIFO_WORDS (USB_RAM_SIZE / 4)
 
-/* Allocate FIFO memory evenly between the FIFOs */
-#define FIFO_EP_WORDS (FIFO_WORDS / FIFO_NUM)
+/* Split the TX FIFOs into large and small buckets */
+#define FIFO_NUM_LG	2		// large mps eps */
+#define FIFO_NUM_SM	(FIFO_NUM - FIFO_NUM_LG - 1)	// small mps eps (leave 1 for RX)
```
Comment (djcb) [inline drivers/usb/device/usb_dc_stm32.c:117]: Upstream doesn't like `//` comments

File: subsys/usb/class/Kconfig.msc [hunk 3]
```diff
@@ -46,8 +46,8 @@ config MASS_STORAGE_INQ_REVISION
 
 config MASS_STORAGE_BULK_EP_MPS
 	int
```
Comment (nickstoughton) [inline subsys/usb/class/Kconfig.msc:48]: add a prompt

**Review by djcb — COMMENTED** [review]
LGTM!

**Pull request merged**